controllers/utils/predicate.go

```diff
@@ -54,16 +56,21 @@ func CreateRequests(client runtimeclient.Client) handler.EventHandler {
 			if err := client.List(context.TODO(), &vgList); err != nil {
 				return []ctrl.Request{}
 			}
-			// TODO CSI-5437 - add a label selector check to the VolumeGroup to filter the list
 			// Create a reconcile request for each matching VolumeGroup.
-			requests := make([]ctrl.Request, len(vgList.Items))
+			var requests []ctrl.Request
 			for _, vg := range vgList.Items {
-				requests = append(requests, ctrl.Request{
-					NamespacedName: types.NamespacedName{
-						Namespace: vg.Namespace,
-						Name:      vg.Name,
-					},
-				})
+				selector, err := metav1.LabelSelectorAsSelector(vg.Spec.Source.Selector)
```

Comment:
will it work on VGs without a selector field/Source field?

Reply:
selector is mandatory field..